client/main.lua

```diff
@@ -0,0 +1,233 @@
+-- Type, Required, UsePersistent
+-- All type validation needed
+local NOTI_TYPES <const> = {
+  ["style"] = { "string", false },
+  ["duration"] = { "number", nil, true },
+  ["sound"] = { "boolean" },
+  ["position"] = { "table" },
+  ["image"] = { "string" },
+  ["icon"] = { "string" }
+}
```

Comment:
This, in my eyes, is not very readable. Especially when doing the index accessing in the verifyTypes call. It could be cleaned up by having these be named sub-properties. Also, avoid bracket notation on string indices not containing special characters.